tests/parametric/test_trace_sampling.py

```diff
@@ -77,6 +77,29 @@ def test_trace_dropped_by_trace_sampling_rule(self, test_agent, test_library):
         assert span["metrics"].get(SAMPLING_PRIORITY_KEY) == -1
         assert span["metrics"].get(SAMPLING_RULE_PRIORITY_RATE) == 0.0
 
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                "DD_TRACE_SAMPLE_RATE": 1,
+                "DD_TRACE_SAMPLING_RULES_FORMAT": "glob",
+                "DD_TRACE_SAMPLING_RULES": json.dumps(
+                    [{"service": "webserver", "resource": "drop-me", "sample_rate": 0}]
+                ),
+            }
+        ],
+    )
+    def test_trace_kept_in_spite_trace_sampling_rule(self, test_agent, test_library):
+        """Test that a trace is being kept with manual.keep depite of the matching defined trace sampling rule"""
+        with test_library:
+            with test_library.start_span(name="web.request", service="webserver") as s1:
+                s1.set_metric("sampling.priority", 2)
```

Comment:
This way of setting sampling priority by users was deprecated a long time ago in favor of setting `"manual.keep"`.